scripts/update-examples-reference-in-docs.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -euo pipefail
+
+: "${1?"Tag of new release must be provided"}"
+
+FILE_PATH="./website/docs/index.html.markdown"
+RELEASE_TAG=$1
+
+# Define the old URL pattern and new URL
+OLD_URL_PATTERN="\[example configurations\](https:\/\/github.com\/mongodb\/terraform-provider-mongodbatlas\/tree\/[a-zA-Z0-9._-]*\/examples)"
+NEW_URL="\[example configurations\](https:\/\/github.com\/mongodb\/terraform-provider-mongodbatlas\/tree\/$RELEASE_TAG\/examples)"
+
+
+TMP_FILE_NAME="docs.tmp"
```

Comment:
[nit] Curious why we are not using "in-place" (`-I` flag)?

Reply:
Went for this approach initially but the implementation of sed in macOS vs Linux handles the `-i` parameter differently. Opted for having same behaviour in CI vs running locally by simplifying how the command is invoked.